### ISSUE TITLE
Export Meta and Configs

### ISSUE THREAD

**Issue by juancarlospaco** [PyAr/fades#127]
- Research and implement a way to Export a single **File** UTF-8 Meta-Data and Configs parsed by Fades actual magic comments and configs, think: 
```python
import fades
fades.export_meta( "my_fades_configs.json" )  # if str arg is passed, then write to that file.
fades.export_meta( file_like_python_object )  # if object arg is passed, then write to that object (object supports .read() and .write()).
```
- Research and implement a way to Export a single **String** UTF-8 Meta-Data and Configs parsed by Fades actual magic comments and configs, think:
```python
import fades
fades.export_meta()  # No args passed, then return the string.
```
- Fades already parses its Data for its magic comments, and I think those probably are JSON-serializable.
- This Export can be a shortcut to implement https://github.com/PyAr/fades/issues/87 *(This is the Import of same data of magic comments)*

---

**Reason:**
The idea is to Fix https://github.com/PyAr/fades/issues/87 and allow third party interoperability, 
for example to use it on Configuration, [Documentation](https://github.com/juancarlospaco/dookumentation#dookumentation) and having Configuration+Documentation will help Debugging.

Enables Documentation to have some Human Description of **what Fades is actually doing for you ?**:
```
Fades will try to install: 
- Requests version 666 from PyPI from a comment found on Line 3.
- BeautifulSoup4 version 4 from git+https://github.com/user/example.git from a comment found on Line 15.
- Six version >1 but <5 from PyPI from a comment found on Line 22.

Fades will create a new VirtualEnv at: "/home/username/.fades/foo/bar/baz" with Python 3.5.

 . . .
```

---

**Usecase already implemented:**
For example this is what [PyLama](https://github.com/klen/pylama#run-pylama-from-python-code) exports ([PrettyPrinted](https://gist.github.com/juancarlospaco/358bcefc7df07bdc6b80#file-pretty_json-py)):
```json
"pylama": [
    {
        "col":                 1,

        "filename":            "templar.py",

        "linter":              "pyflakes",

        "lnum":                8,

        "number":              "W0611",

        "text":                "W0611 'requests' imported but unused [pyflakes]",

        "type":                "E"
    },

    {
        "col":                 1,

        "filename":            "templar.py",

        "linter":              "pyflakes",

        "lnum":                72,

        "number":              "E0602",

        "text":                "E0602 undefined name 're' [pyflakes]",

        "type":                "E"
    }
]

```

---

**TL;DR:**

- Dump everything into a JSON or something like that, so we can Read/Write to/from it.

:smile_cat: :boom:

**Comment by facundobatista** [comment]
Some details:

- this would be the first time "fades" is used as a module, not a executable

- I don't see the relationship with #87, maybe I'm not fully understanding how you're mixing config and what would be installed.

- it may be useful to produce a json with "what the dependencies are after finding what where", but I prefer to have a real life use case for this, and not "let's produce a lot of code for something that may be useful"

**Comment by juancarlospaco** [comment]
Or
```
fades --export-meta     # std out
fades --export-meta "/path/to/file.json"
```
But I dunno if thats more simple to implement, but is also an alternative.


The output definitely needs design.
:smile_cat:

**Comment by facundobatista** [comment]
Yes, we could use a parameter... but if your plan is to use it from another Python program, just make it a module functionality! or both ;)

However, the central point here is what to produce and if it would be useful to anybody...

**Comment by juancarlospaco** [comment]
You can still use subprocess module, feel free to design that, and also the serialized data.


It should be useful to you, thats why I was saying you can implement the load config from there too,
load a config from a serialized file, dump a config to a serialized file. 

When you dump, dump the config and the actual data fades has parsed, 
when you load, load the config, ignore the data.

**Comment by facundobatista** [comment]
After talking about this in IRC we decided that there's nobody currently needing this, so code to make this happen only will produce new bugs.

Happy to start talking about this again in the future if an actual needs arises.

Thanks for your time!

**Comment by juancarlospaco** [comment]
:+1:
